src/pybind/mgr/mds_autoscaler/module.py

```diff
@@ -0,0 +1,142 @@
+"""
+Automatically scale MDSs based on status of the file-system using the FSMap
+"""
+
+import logging
+from typing import Optional, List
+from mgr_module import MgrModule
+from ceph.deployment.service_spec import ServiceSpec, PlacementSpec
+import orchestrator
+import json
+
+log = logging.getLogger(__name__)
+
+"""
+Some terminology is made up for the purposes of this module:
+
+ - "grow/shrink" - increase/decrease the MDSs in the system
+"""
+
+
+class MDSAutoscaler(orchestrator.OrchestratorClientMixin, MgrModule):
+    """
+    MDS autoscaler.
+    """
+    def __init__(self, *args, **kwargs):
+        super(MDSAutoscaler, self).__init__(*args, **kwargs)
+        self.set_mgr(self)
+        self.fs_map: Optional[dict] = None
+        self.old_notify_type = "NONE"
+
+    def get_service(self, fs_name) -> Optional[List[orchestrator.ServiceDescription]]:
+        result = None
+        try:
+            #service = "mds.{}".format(fs_name)
+            service = "mds.newb"
+            args = ()
+            completion = self.describe_service(service_type='mds',
+                                               service_name=service,
+                                               refresh=True)
+            self._orchestrator_wait([completion])
+            orchestrator.raise_if_exception(completion)
+            result = completion.result
+        except Exception as e:
+            self.log.exception("{}: exception while fetching service info"
+                               .format(e))
+            pass
+        self.log.info("service info:{}".format(result[0]))
+        return result
+
+    def update_daemon_count(self, fs_name, abscount) -> Optional[ServiceSpec]:
+        svclist = self.get_service(fs_name)
+        if svclist is None or len(svclist) == 0:
+            self.log.warn("Failed to fetch MDS service details for {}"
+                           .format(fs_name))
+            return None
+
+        svc = svclist[0]
+        self.log.info("newsvc:{}".format(svc))
+        self.log.info("newsvc.spec:{}".format(svc.spec))
+        self.log.info("new count:{}".format(abscount))
+
+        hosts = []
+        for h in svc.spec.placement.hosts:
+            hosts.append(h.hostname)
+        ps = PlacementSpec(hosts=hosts, count=abscount)
```

Comment:
shouldn't you also be copying over `label` and `host_pattern` attributes also?

Can you just copy over svc.spec.placement,
```
ps = copy.copy(svc.spec.placement)
ps.count = abscount
```

Reply:
fixed